plan/aggregation_push_down.go

```diff
@@ -291,20 +300,34 @@ func (a *aggPushDownSolver) pushAggCrossUnion(agg *Aggregation, unionSchema *exp
 		newAgg.GroupByItems = append(newAgg.GroupByItems, newExpr)
 	}
 	newAgg.collectGroupByColumns()
+	tmpSchema := expression.NewSchema(newAgg.groupByCols...)
+	for _, key := range unionChild.Schema().Keys {
+		if tmpSchema.ColumnsIndices(key) != nil {
+			proj, err := convertAggToProj(newAgg, a.ctx, a.alloc)
+			if err != nil {
+				return nil, errors.Trace(err)
+			}
+			proj.SetChildren(unionChild)
+			return proj, nil
```

Comment:
For example, for union distinct, currently the pushed agg in union is some thing like `select a, b, c, a, b, c from t group by a, b, c`. If we just return the union child directly, this will cause error in executor phase.